test/test_matcher.cpp

```diff
@@ -36,14 +36,15 @@ BOOST_AUTO_TEST_CASE(ref_to_int_and_int_can_be_compared)
 namespace {
 struct fixture
 {
-    fixture() : text("same text"), actual(text.c_str())
+    fixture() : text("same text"), actual(text.c_str()), null_str(nullptr)
     {
         const char* static_string = "same text";
         BOOST_REQUIRE(actual != static_string);
         BOOST_REQUIRE(actual == std::string(static_string));
     }
     std::string text;
     const char* actual;
+    const char* null_str;
```

Comment:
I’m not sure this needs to be factor out in the fixture, given it’s only used once?

Reply:
True. Now that I look at this again I think I was trying too hard to fit into the existing pattern of these tests. I agree that it would be simpler just to define this in the one test that uses it.